### ISSUE TITLE
[Query] How confluent-kafka-go consumer works for multiple topics

### ISSUE THREAD

**Issue by anandkolli** [confluentinc/confluent-kafka-go#337]
Description
===========
Hello,

I have consumer which has subscribed for multiple topics say [KPI, Alarm, Data]
On KPI topic I see a lot of traffic and consumer is processing.

During the same time If I produce a message on Data topic, the consumer never reads that message and it keeps processing the KPIs
I have multiple goroutines running for KPI, Alarm, and Data.
and there are no deadlocks, it's just that message is not delivered to the consumer.

And the message is delivered to Kafka, I have cross verified using Kafkacat.

Below is the snippet from my code
```
        go func() {
                for {
                        data := c.consumer.Poll(defaultPollTime)
                        if data != nil {
                                switch dataType := data.(type) {
                                case *kafka.Message:
                                        // Populate partition info.
                                        partitionInfo := EOFEvent{dataType.TopicPartition.Topic, dataType.TopicPartition.Partition,
                                                int64(dataType.TopicPartition.Offset), dataType.TopicPartition.Error}
                                        // Populate User Message.
                                        usrMsg := UserMessage{partitionInfo, dataType.Value, dataType.Key, dataType.Headers}
                                        // Call user provided function.
                                        handler(MsgData(&usrMsg))
                                case kafka.Error:
                                        // Retrieve error code and string.
                                        err := ErrorEvent{int(dataType.Code()), dataType.String()}
                                        // Call user provided function.
                                        handler(MsgData(&err))
                                case kafka.PartitionEOF:
                                        // Populate partition info.
                                        partitionData := EOFEvent{dataType.Topic, dataType.Partition, int64(dataType.Offset), dataType.Error}
                                        // Call user provided function.
                                        handler(MsgData(&partitionData))
                                }
                        }
```
Question is how consumer Poll works when subscribed to multiple topics?

or Do we need to create a separate goroutine per topic and then Poll?

Checklist
=========
Please provide the following information:

 - [ ] confluent-kafka-go and librdkafka version (`LibraryVersion()`):v0.11.6 and librdkafka is latest
 - [ ] Apache Kafka broker version:inter.broker.protocol.version = 1.0-IV0
 - [ ] Client configuration: `ConfigMap{...}`
 - [ ] Operating system: Ubuntu 16.04
 - [ ] Provide client logs (with `"debug": ".."` as necessary)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

Any input or suggestion will be very helpful.

Thank you,
Anand

**Comment by aelous** [comment]
I also encountered the same problem.

**Comment by edenhill** [comment]
See https://github.com/edenhill/librdkafka/wiki/FAQ#how-are-partitions-fetched

TLDR: reduce `fetch.message.max.bytes`
